api/src/test/java/org/openmrs/api/CohortServiceTest.java

```diff
@@ -521,8 +525,92 @@ public void removePatientFromCohort_shouldSaveCohortAfterRemovingPatient() throw
 		// make a patient, add it using the method
 		Patient patientToAddThenRemove = new Patient(4);
 		service.addPatientToCohort(service.getCohort(2), patientToAddThenRemove);
-		assertTrue(service.getCohort(2).contains(patientToAddThenRemove));
+		assertTrue(service.getCohort(2).contains(patientToAddThenRemove.getPatientId()));
 		service.removePatientFromCohort(service.getCohort(2), patientToAddThenRemove);
-		assertFalse(service.getCohort(2).contains(patientToAddThenRemove));
+		List<CohortMembership> memberList = service.getCohort(2)
+				.getMembers().stream()
+				.filter(m -> m.getPatient().getPatientId().equals(patientToAddThenRemove.getPatientId()))
+				.collect(Collectors.toList());
+		CohortMembership memberWithPatientToRemove = memberList.get(0);
+        assertNotNull(memberWithPatientToRemove.getEndDate());
+	}
+
+	/**
+	 * @verifies {@link CohortService#addMembershipToCohort(Cohort, CohortMembership)}
+	 * test = should add membership to cohort
+	 */
+	@Test
+	@Verifies(value = "should add membership to cohort", method = "addMembershipToCohort(Cohort, CohortMembership)")
+	public void addMembershipToCohort_shouldAddMembershipToCohort() throws Exception {
+		executeDataSet(COHORT_XML);
+
```

Comment:
Would nice to have a sanity check that the cohort doesn't contain an active membership for the patient before